compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -872,6 +882,47 @@ public void visit(BLangSimpleVariable varNode) {
         transferForkFlag(varNode);
     }
 
+    private void analyzeModuleConfigurableAmbiguity(BLangPackage pkgNode) {
+        Set<BVarSymbol> configVars = symResolver.getConfigVarSymbols(pkgNode.symbol);
+        populateModuleKeys(configVars);
+        for (BVarSymbol variable : configVars) {
+            validateMapConfigVariable(variable.pkgID.orgName.value + "." + variable.pkgID.name.value + "."
+                    + variable.name.value, variable);
+            if (variable.pkgID.getOrgName().equals(rootPackage.getOrgName())) {
+                validateMapConfigVariable(variable.pkgID.name.value + "." + variable.name.value, variable);
+                if (variable.pkgID.getName().equals(rootPackage.getName()) &&
+                        !(variable.name.equals(variable.pkgID.name))) {
+                    validateMapConfigVariable(variable.name.value, variable);
+                }
+            }
+        }
+    }
+
+    private void populateModuleKeys(Set<BVarSymbol> configVars) {
+        for (BVarSymbol variable : configVars) {
+            configKeys.add(variable.pkgID.orgName.value + "." + variable.pkgID.name.value);
+            if (!variable.pkgID.getOrgName().equals(rootPackage.getOrgName())) {
+                break;
+            }
+            configKeys.add(variable.pkgID.name.value);
+        }
+    }
+
+    private void validateMapConfigVariable(String configKey, BVarSymbol variable) {
+        if (configKeys.contains(configKey) && isSubTypeOfMapping(variable.type) && !invalidVars.contains(variable)) {
+            invalidVars.add(variable);
+            dlog.error(variable.pos, DiagnosticErrorCode.CONFIGURABLE_VARIABLE_MODULE_AMBIGUITY_FOUND,
+                    variable.name.value);
+        }
+    }
+
+   private boolean isSubTypeOfMapping(BType type) {
```

Comment:
This has to be part of the types.isSubTypeOfMapping itself. So we can remove this.